main.py

```diff
@@ -80,6 +80,14 @@ def setDropdownAttr(widget, var): self.settings.settings[var] = widget.currentTe
 		def getDropdownAttr(widget, var): widget.setCurrentText(self.settings.settings[var])
 		def updateLaunchCmd(): self.invocationPreview.setPlainText(Xqemu.launchCmdToString(Xqemu.generateLaunchCmd(self.settings, True)))
 
+		def updateControllerUi():
+			# Update all four controllers
+			for i in [1, 2, 3, 4]:
```

Comment:
nit: Why do we use 1-4 not 0-3?
nit: It's more idiomatic to use Python's `range` built-in

Reply:
The UI uses 1-4 (which I think is more natural for the end-user), so I tried to keep it instead of "translating" to 0-3 to avoid mix-ups, should the code be edited in the future.
I used `for i in range(1, 5)`, on which JFR rightfully remarked that a five appearing in this context is not ideal, so I switched to this, which is slightly less pythonic, but shows what's going on more clearly, I think.

Reply:
I think this is a good solution as it's controller ports 1-4 which is how they are labelled on the Xbox case.

Reply:
Fair enough :)